File: calzados-frontend/src/app/core/services/comprobante.service.ts
// src/app/core/services/comprobante.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../models/api-response.model';
import { ComprobanteRequest, ComprobanteResponse, TipoComprobante } from '../models/index';

@Injectable({ providedIn: 'root' })
export class ComprobanteService {
  private api = `${environment.apiUrl}/comprobantes`;
  constructor(private http: HttpClient) {}

  emitir(body: ComprobanteRequest): Observable<ComprobanteResponse> {
    return this.http.post<ApiResponse<ComprobanteResponse>>(this.api, body).pipe(map(r => r.data));
  }

  obtener(id: number): Observable<ComprobanteResponse> {
    return this.http.get<ApiResponse<ComprobanteResponse>>(`${this.api}/${id}`).pipe(map(r => r.data));
  }

  obtenerPorSerie(serie: string): Observable<ComprobanteResponse> {
    return this.http.get<ApiResponse<ComprobanteResponse>>(`${this.api}/serie/${serie}`).pipe(map(r => r.data));
  }

  obtenerPorVenta(ventaId: number): Observable<ComprobanteResponse> {
    return this.http.get<ApiResponse<ComprobanteResponse>>(`${this.api}/venta/${ventaId}`).pipe(map(r => r.data));
  }

  listarPorFecha(inicio: string, fin: string, tipo?: TipoComprobante): Observable<ComprobanteResponse[]> {
    let params = new HttpParams().set('inicio', inicio).set('fin', fin);
    if (tipo) params = params.set('tipo', tipo);
    return this.http.get<ApiResponse<ComprobanteResponse[]>>(this.api, { params }).pipe(map(r => r.data));
  }

  listarPorCliente(clienteId: number): Observable<ComprobanteResponse[]> {
    return this.http.get<ApiResponse<ComprobanteResponse[]>>(`${this.api}/cliente/${clienteId}`)
      .pipe(map(r => r.data));
  }
}

File: calzados-pos/src/main/java/com/pos/calzados/service/ComprobanteService.java
package com.pos.calzados.service;

import com.pos.calzados.dto.request.ComprobanteRequest;
import com.pos.calzados.dto.response.ComprobanteResponse;
import com.pos.calzados.entity.Rol;
import com.pos.calzados.entity.TipoComprobante;

import java.time.LocalDate;
import java.util.List;

public interface ComprobanteService {
    ComprobanteResponse emitir(ComprobanteRequest request);
    ComprobanteResponse obtenerPorId(Long id, Long userId, Rol rol);
    ComprobanteResponse obtenerPorSerie(String serie, Long userId, Rol rol);
    ComprobanteResponse obtenerPorVentaId(Long ventaId, Long userId, Rol rol);
    List<ComprobanteResponse> listarPorFecha(LocalDate inicio, LocalDate fin, TipoComprobante tipo, Long userId, Rol rol);
    List<ComprobanteResponse> listarPorCliente(Long clienteId, Long userId, Rol rol);
}
